### DISCUSSION TITLE
latexstring in markdown not working

### DISCUSSION THREAD

**Discussion by dorn-gerhard** [fonsp/Pluto.jl#1215] [Pluto development] [answered]
I experienced a change in the behavior of **latexstring in markdown** environment in pluto.
Basically it does not work anymore.

I would like to use Markdown + Variables + LaTeX
The best workaround so far was to use
a)  `md""" $(latexstring("y =")) $(variable) $(latexstring("x+d" )) """`
since the Latex-Markdown variant with $\int...$ did not work with preceeding variables:
b) `md""" $y$ = $(variable) $x + d$ """`

Unfortunately latexstring does not work in the markdown environment anymore. 
Does this depend on the mentioned changes in the LaTeXStrings package or in Pluto #359 #488  #1164?

![grafik](https://user-images.githubusercontent.com/67096719/120459351-6876e600-c398-11eb-9425-8c50f85ddd3c.png)

I am using LaTeXStrings v1.2.1 and Pluto v0.14.7 and tested on two machines / installations

What will be the future strategy or best practise to use Markdown + Variables + LaTeX?

**Comment by holomorphism** [comment, marked as answer]
I'm currently doing the following:

1. First paste the following code into a cell:

   ```julia
   begin
   	function m(xs...; env=:none)
   		if env==:none
   			_latex(xs...) |> Markdown.LaTeX
   		else
   			wrapenv(env, xs...) |> Markdown.LaTeX
   		end
   	end
   	mm(xs...; kw...) = m(xs...; kw...) |> Markdown.MD
   	wrapenv(e, xs...) = _latex("\\begin{$e}\n", xs..., "\n\\end{$e}")
   
   	align(xs...) = m(env="align", xs...)
   	gather(xs...) = m(env="gather", xs...)
   
   	_latex(xs...) = replace(join(_latex.(xs), " "), r"^ +|(?<!\\) +$"m => "")
   
   	if @isdefined SymPy
   		_latex(x::SymPy.SymbolicObject) = sympy.latex(x)  # mode="plain"
   	end
   
   	_latex(x::AbstractString) = strip(x,'$')  # for LaTeXString
   
   	function _latex(x::Rational)
   		n, d = numerator(x), denominator(x)
   		d == 1 ? string(n) : (n<0 ? "-" : "") * "\\frac{$(abs(n))}{$d}"
   	end
   
   	function _latex(x)
   		if showable(MIME("text/latex"), x)
   			repr("text/latex", x)
   		else
   			string(x)
   		end
   	end
   
   	function _latex(x::Union{AbstractVector,AbstractMatrix})
   		rows = map(eachrow(x)) do row join(_latex.(row),"&") end
   		inner = join(rows, "\\\\")
   		"\\left[\\begin{array}{$("c"^size(x,2))}" * inner * "\\end{array}\\right]"
   	end
   	
   	NL = "\\\\\n"
   end;
   ```

2. Then you can combine the latex formulas in markdown as follows:

   ![image](https://user-images.githubusercontent.com/3076585/120878779-9eff6b80-c5f9-11eb-92f7-a75f1151184d.png)


cf. https://github.com/fonsp/Pluto.jl/issues/488#issuecomment-735393368

**Comment by fonsp** [comment]
More discussion in:
https://github.com/fonsp/Pluto.jl/issues/1218
and
https://github.com/fonsp/Pluto.jl/issues/1237
